panel/esm.py

```diff
@@ -91,10 +92,17 @@ def __init__(self, **params):
         self._event_callbacks = defaultdict(list)
 
     def _render_esm(self):
-        if isinstance(self._esm, pathlib.PurePath):
-            esm = self._esm.read_text()
-        else:
-            esm = self._esm
+        esm = self._esm
+        if isinstance(esm, pathlib.PurePath):
+            esm = esm.read_text(encoding='utf-8')
+        elif esm.endswith(('.js', '.jsx', '.ts', '.tsx')):
+            try:
```

Comment:
It now reads files given by strings. But it does not add hot reloading.

Reply:
Hot reloading works for me at least, if I use `--autoreload` anyway.

Reply:
Oh nvm, I get your point, we're not yet resolving the path correctly in the `_watch_esm` call.

Reply:
works for me now.